polaris-core/src/main/java/org/apache/polaris/core/storage/s3compatible/S3CompatibleCredentialsStorageIntegration.java

```diff
@@ -0,0 +1,164 @@
+    // do not care about regions
+    stsBuilder.endpointOverride(
+        URI.create(
+            s3storageConfig
+                .getS3Endpoint())); // S3Compatible - AWS STS endpoint is unique and different from
+    // the S3 Endpoint
+    stsBuilder.credentialsProvider(
+        StaticCredentialsProvider.create(
+            AwsBasicCredentials.create(
+                s3storageConfig.getS3CredentialsCatalogAccessKeyId(),
+                s3storageConfig.getS3CredentialsCatalogSecretAccessKey())));
+    this.stsClient = stsBuilder.build();
+    LOGGER.debug("S3Compatible - stsClient successfully built");
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public EnumMap<PolarisCredentialProperty, String> getSubscopedCreds(
+      @NotNull PolarisDiagnostics diagnostics,
+      @NotNull S3CompatibleStorageConfigurationInfo storageConfig,
+      boolean allowListOperation,
+      @NotNull Set<String> allowedReadLocations,
+      @NotNull Set<String> allowedWriteLocations) {
+
+    LOGGER.debug("S3Compatible - getSubscopedCreds - applying credential strategy");
+    EnumMap<PolarisCredentialProperty, String> propertiesMap =
+        new EnumMap<>(PolarisCredentialProperty.class);
+    propertiesMap.put(PolarisCredentialProperty.AWS_ENDPOINT, storageConfig.getS3Endpoint());
+    propertiesMap.put(
+        PolarisCredentialProperty.AWS_PATH_STYLE_ACCESS,
+        storageConfig.getS3PathStyleAccess().toString());
+
+    switch (storageConfig.getCredsVendingStrategy()) {
+      case KEYS_SAME_AS_CATALOG:
+        if (storageConfig.getCredsCatalogAndClientStrategy()
+            == CredsCatalogAndClientStrategyEnum.ENV_VAR_NAME) {
+          String cai = System.getenv(storageConfig.getS3CredentialsCatalogAccessKeyId());
+          String cas = System.getenv(storageConfig.getS3CredentialsCatalogSecretAccessKey());
+          if (cai == null || cas == null) {
+            throw new IllegalArgumentException(
+                "Expect valid environment variable for catalog keys");
+          } else {
+            propertiesMap.put(PolarisCredentialProperty.AWS_KEY_ID, cai);
+            propertiesMap.put(PolarisCredentialProperty.AWS_SECRET_KEY, cas);
+          }
+        } else {
+          propertiesMap.put(
+              PolarisCredentialProperty.AWS_KEY_ID,
+              storageConfig.getS3CredentialsCatalogAccessKeyId());
+          propertiesMap.put(
+              PolarisCredentialProperty.AWS_SECRET_KEY,
+              storageConfig.getS3CredentialsCatalogSecretAccessKey());
+        }
+        break;
+
+      case KEYS_DEDICATED_TO_CLIENT:
+        if (storageConfig.getCredsCatalogAndClientStrategy()
+            == CredsCatalogAndClientStrategyEnum.ENV_VAR_NAME) {
+          String cli = System.getenv(storageConfig.getS3CredentialsClientAccessKeyId());
+          String cls = System.getenv(storageConfig.getS3CredentialsClientSecretAccessKey());
```

Comment:
Seems you could rely on the `DefaultCredentialsProvider` and maybe allow profiles to be specified? This would allow for env variables, but also file configuration or other means of retrieving credentials.

Reply:
Do not find how to conciliate this with createCatalog() REST API... And not happy with the idea to let this credentials catalog's fully and exclusively at Polaris service level.
Bad compromise ?

Reply:
DefaultCredentialsProvider : done